File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.java
import java.util.*;
import java.util.stream.*;


class Solution {

    public boolean solution(String[] phone_book) {
        Set<String> phoneNumbers = new HashSet<>(Arrays.asList(phone_book));

        for (final String phone_number : phone_book) {
            for (int i = 1; i < phone_number.length(); i++) {
                if (phoneNumbers.contains(phone_number.substring(0, i))) {
                    return false;
                }
            }
        }
        return true;
    }
}

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.py
def solution(phone_book):
    phone_book.sort()
    curr = phone_book[0]
    
    # 12 13
    
    for num in phone_book[1:]:
        if curr != num[:len(curr)]:
            curr = num
        else:
            return False
        
    return True
